src/redux.js

```diff
@@ -1293,3 +1293,9 @@ function areArraysEqual(array1, array2) {
     })
   );
 }
+
+export function isUserUploadedDataset(state) {
+  // The csvfile for internally curated datasets are strings; those uploaded by
+  // users are objects. Use data type as a proxy to know which case we're in.
+  return typeof state.csvfile === 'object' && state.csvfile !== null;
```

Comment:
Overloading data type to have a deeper meaning, like this, makes me slightly nervous; it's often clearer to have an explicit value that represents how it was uploaded, especially as a way of future-proofing.  I guess the helper function does make things fairly clear, but I'm curious if you also have this concern...

Reply:
It makes me slightly nervous, too, which is why I added the comment to make it clearer that it's kind of hacky. Any time I find myself saying something is proxy, I usually mean it's a shortcut which has it's pros and cons 😄  To me this felt like the simpler and therefore safer choice at least in the short term. But I could be convinced to add a new piece of state,  that gets set in SelectDataset if you think this is a likely hole our future selves will stumble into.

Reply:
Let's go ahead with this for now, and we'll see how it feels later :)